app/assets/javascripts/analytics/ecommerce.js

```diff
@@ -0,0 +1,62 @@
+(function () {
+  "use strict";
+  window.GOVUK = window.GOVUK || {};
+
+  var Ecommerce = function (config) {
+    this.init = function (element) {
+      var self = this;
+      var ecommerceRows = element.find('[data-ecommerce-row]');
+      var startPosition = parseInt(element.data('ecommerce-start-index'));
+
+      ecommerceRows.each(function(index, ecommerceRow) {
+        var $ecommerceRow = $(ecommerceRow);
+
+        var contentId = $ecommerceRow.attr('data-ecommerce-content-id'),
+          path = $ecommerceRow.attr('data-ecommerce-path');
+
+        self._sendImpression($ecommerceRow, contentId, path, index + startPosition);
+        self._trackProductClick($ecommerceRow, contentId, path, index + startPosition);
+      });
+    }
+
+    this._sendImpression = function (_row, contentId, path, position) {
+      // We only send the id to GA as additional product data is linked when it is uploaded.
+      // This approach is taken to avoid the GA data packet exceeding the 8k limit
+      ga('ec:addImpression', {
```

Comment:
I spoke with @nickcolley  when we started on this work and the decision was made to not have this code in the `frontend toolkit` until we could prove that it worked/we would be keeping it. It's for this reason that we are accessing `ga` directly here. If we are happy with the code we can move this whole file into the frontend toolkit which will then mirror the remainder of the codebase.